dehydrated

```diff
@@ -450,18 +450,24 @@ sign_csr() {
 
   local idx=0
   if [[ -n "${ZSH_VERSION:-}" ]]; then
-    local -A challenge_uris challenge_tokens keyauths deploy_args
+    local -A challenge_uris challenge_tokens keyauths deploy_args already_valid
   else
-    local -a challenge_uris challenge_tokens keyauths deploy_args
+    local -a challenge_uris challenge_tokens keyauths deploy_args already_valid
   fi
 
   # Request challenges
   for altname in ${altnames}; do
     # Ask the acme-server for new challenge token and extract them from the resulting json block
     echo " + Requesting challenge for ${altname}..."
     response="$(signed_request "${CA_NEW_AUTHZ}" '{"resource": "new-authz", "identifier": {"type": "dns", "value": "'"${altname}"'"}}' | clean_json)"
+    if [[ $response == *"\"status\": \"valid\""* ]]; then
```

Comment:
There are several `"status": ...` fields in the response, but probably it works out ok if any of them is `valid`. I assume the toplevel `status` would be the one we really are interested in.

`$reponse` should be quoted in case it contains spaces.